packages/api/app/Controllers/Http/TechnologyController.js

```diff
@@ -167,20 +170,51 @@ class TechnologyController {
 		return response.status(200).send({ success: true });
 	}
 
-	async syncronizeUsers(trx, users, technology, detach = false) {
+	async syncronizeUsers(trx, users, technology, detach = false, provisionUser = false) {
 		if (detach) {
 			await technology.users().detach(null, null, trx);
 		}
-		const usersId = users.map((item) => item.userId);
-		const userMap = new Map(users.map((user) => [user.userId, user.role]));
+
+		const usersToFind = [];
+		let resultUsers = [];
+
+		users.forEach(async (user) => {
+			const { id, email } = user;
+			if (id) {
+				resultUsers.push(user);
+			} else {
+				usersToFind.push(
+					provisionUser
+						? User.findOrCreate(
+								{ email },
+								{ ...user, password: randtoken.generate(12), status: 'invited' },
+						  )
+						: User.findBy('email', email),
+				);
+			}
+		});
+
+		// Returns the users found in the db, the provisioned ones and null (in case the user was not found and not provisioned)
+		const foundUsers = await Promise.all(usersToFind);
+		resultUsers = [...resultUsers, ...foundUsers.filter((user) => user !== null)];
```

Comment:
```suggestion
		resultUsers = [...resultUsers, ...foundUsers.filter(Boolean)];
```
Truque simples para remover nulls de um array

Reply:
:100: